### PR TITLE
Rework GPflow introduction.

### PR DESCRIPTION
**Pull request by jesnie**
Rewrite of GPflow documentation landing page and "getting started" documentation.

A rendered version of this is available at https://gpflow.github.io/GPflow/rework/index.html

### CODE REVIEW

File: doc/sphinx/getting_started.rst [hunk 1]
```diff
@@ -0,0 +1,28 @@
+Getting Started
+===============
+
+This section aims to give you the knowledge necessary to use GPflow on small-to-medium projects,
+without necessarily going too much into the mathematical and technical details. We do not try to
+teach the theory behind Gaussian Processes. If you need a deeper understanding of the technical
+details or advanced features of GPflow please see our `User Guide`_.
```
Comment (frgsimpson) [inline doc/sphinx/getting_started.rst:7]:
```suggestion
details or advanced features of GPflow please see our `User Guide`.
```

Comment (jesnie) [reply doc/sphinx/getting_started.rst]: Fergus, that `_` turns it into a link. It's not a mistake. :-)

Comment (frgsimpson) [reply doc/sphinx/getting_started.rst]: Hah ok good to know. :) I could have sworn I saw it rendered as a '_' in the browser, but it looks fine now. The only small thing I spot on the 'Getting Started' page now is: "Case Study: \(CO_2\) Levels"

**Review by frgsimpson — COMMENTED** [review]
Looks great, just some small suggestions!

File: doc/sphinx/index.rst [hunk 2]
```diff
@@ -1,24 +1,107 @@
-GPflow Documentation
-==================================
+GPflow
+======
 
-.. toctree::
-   :maxdepth: 1
-   :caption: Intro:
+GPflow is a package for building Gaussian Process models in python, using `TensorFlow
+<http://www.tensorflow.org>`_. Gaussian Processes are a kind of supervised learning models.
```
Comment (frgsimpson) [inline doc/sphinx/index.rst:5]:
```suggestion
<http://www.tensorflow.org>`_. A Gaussian Process is a kind of supervised learning model.
```
not sure if this is the best rewording, but currently there is a bit of a mismatch between singular & plural.

**Review by frgsimpson — COMMENTED** [review]
Looks good! I'd suggest the landing page should make it clear which place to go if the user is wanting help to implement a particular model like SGPR or SVGP.

File: doc/sphinx/index.rst (hunk 2 above)
Comment (jesnie) [reply doc/sphinx/index.rst]: > not sure if this is the best rewording, but currently there is a bit of a mismatch between singular & plural.

Done.

File: doc/sphinx/notebooks/getting_started/mean_functions.pct.py [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+# %%
+model = gpflow.models.GPR(
+    (X, Y),
+    kernel=gpflow.kernels.Matern32(),
+    mean_function=gpflow.functions.Polynomial(2),
+)
+plot_model(model, min_x=-0.1, max_x=4.1)
+
+# %% [markdown]
+# Our model now fits our mean function to the data without any noise, and happily extrapolates the quadtratic function.
```
[the remark this answers is not in the document]
Comment (jesnie) [reply doc/sphinx/notebooks/getting_started/mean_functions.pct.py]: Done.

File: doc/sphinx/notebooks/getting_started/mean_functions.pct.py [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+
+# %% [markdown]
+# ## Interactions with the kernel
+#
+# The shape of $f$, determined by the kernel, is relative to the mean function.
```
[the remark this answers is not in the document]
Comment (jesnie) [reply doc/sphinx/notebooks/getting_started/mean_functions.pct.py]: > The shape of $f$ relative to the mean function is determined by the kernel.

Yeah, good call.

File: doc/sphinx/notebooks/getting_started/mean_functions.pct.py [hunk 5]
```diff
@@ -0,0 +1,201 @@
...
+# Observe:
+
+# %%
+X = np.array([[0.0], [0.1], [0.2], [0.3], [0.4]])
+Y = np.array([[2.0], [1.7], [1.6], [1.7], [2.0]])
+model = gpflow.models.GPR((X, Y), kernel=gpflow.kernels.Matern32())
+plot_model(model, min_x=-0.1, max_x=4.1)
+
+# %% [markdown]
+# As we move away from the data our predictions converge towards 0. Basically, as we move away from the data, the model does not know what to predict, and will revert towards the mean function. This is called ["mean reversal](https://en.wikipedia.org/wiki/Regression_toward_the_mean), and this is the primary use case where having a reasonable mean function can help. In the areas where we have plenty of data, we don't need a mean function.
```
[the remark this answers is not in the document]
Comment (jesnie) [reply doc/sphinx/notebooks/getting_started/mean_functions.pct.py]: Fixed.

File: doc/sphinx/notebooks/getting_started/mean_functions.pct.py [hunk 6]
```diff
@@ -0,0 +1,201 @@
...
+
+# %% [markdown]
+# ## Setting a mean function
+#
+# Our models generally take a `mean_function` parameter that you can set to any instance of [gpflow.functions.MeanFunction](../../api/gpflow/functions/index.html#gpflow.functions.MeanFunction). See our [API documentation](../../api/gpflow/functions/index.html) for a full list of built-in mean functions.
```
[the remark this answers is not in the document]
Comment (jesnie) [reply doc/sphinx/notebooks/getting_started/mean_functions.pct.py]: Fixed.

File: doc/sphinx/notebooks/getting_started/classification_and_other_data_distributions.pct.py [hunk 7]
```diff
@@ -0,0 +1,317 @@
...
+
+import gpflow
+
+# hide: begin
+# %matplotlib inline
+plt.rcParams["figure.figsize"] = (12, 6)
+# hide: end
+
+# %% [markdown]
+# ## The Variational Gaussion Process
```
Comment (frgsimpson) [inline doc/sphinx/notebooks/getting_started/classification_and_other_data_distributions.pct.py:50]:
```suggestion
# ## The Variational Gaussian Process
```

**Review by sc336 — APPROVED** [review]

**Pull request merged**